controller/src/main/java/cern/jarrace/controller/rest/controller/AgentContainerController.java

```diff
@@ -0,0 +1,104 @@
+package cern.jarrace.controller.rest.controller;
+
+import cern.jarrace.controller.domain.Entrypoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * @author timartin
+ * Controller that exposes services to manage {@link Entrypoint}s
+ */
+@RestController
+@RequestMapping("/jarrace")
+public class AgentContainerController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AgentContainerController.class);
+    private static final File DEPLOYMENT_DIR = new File(System.getProperty("java.io.tmpdir"));
+
+    final Map<String, List<Entrypoint>> entrypoints = new HashMap<>();
+    final Map<String, String> paths = new HashMap<>();
+
+    @RequestMapping(value = "/container/deploy/{name}", method = RequestMethod.POST)
```

Comment:
I agree with JC: The client should have no control about the name of the agent. The information has to be in the jar.(at least in the longterm)